pkg/tangy/rpm.go

```diff
@@ -216,6 +234,75 @@ func (t *tangyImpl) RpmRepositoryVersionEnvironmentSearch(ctx context.Context, h
 	return rpms, nil
 }
 
+// RpmRepositoryVersionErrataList List Errat within a repository version, with pagination, and an optional filters
+func (t *tangyImpl) RpmRepositoryVersionErrataList(ctx context.Context, hrefs []string, filterOpts ErrataListFilters, pageOpts PageOptions) ([]ErrataListItem, int, error) {
+	if len(hrefs) == 0 {
+		return []ErrataListItem{}, 0, nil
+	}
+
+	conn, err := t.pool.Acquire(ctx)
+	if err != nil {
+		return nil, 0, err
+	}
+	defer conn.Release()
+
+	if pageOpts.Limit == 0 {
+		pageOpts.Limit = DefaultLimit
+	}
+
+	repoVerMap, err := parseRepositoryVersionHrefsMap(hrefs)
+	if err != nil {
+		return nil, 0, fmt.Errorf("error parsing repository version hrefs: %w", err)
+	}
+
+	countQueryOpen := "select count(*) as total FROM rpm_updaterecord re WHERE re.content_ptr_id IN "
+
+	args := pgx.NamedArgs{
+		"idFilter":       filterOpts.ID,
+		"typeFilter":     filterOpts.Type,
+		"severityFilter": filterOpts.Severity,
+	}
+
+	var filterBytes bytes.Buffer
+	if filterOpts.ID != "" {
+		filterBytes.WriteString(" AND re.id ILIKE CONCAT( '%', @idFilter::text, '%')")
+	}
+	if filterOpts.Type != "" {
+		filterBytes.WriteString(" AND re.type = @typeFilter::text")
+	}
+	if filterOpts.Severity != "" {
+		filterBytes.WriteString(" AND re.severity = @severityFilter::text")
+	}
+	filterQuery := filterBytes.String()
```

Comment:
@jlsherrill  I'm not sure this is needed, I'm not super familiar with pgx. 
Also I would expect the AND re.severity = @severityFilter::text to be an "equal" not "like" comparison. 
But that is what I am seeing.

Reply:
Assuming this is working I think it's okay to concat strings like this. Although it looks like filterBytes is a `bytes.Buffer` then you're converting it to a string. You should probably just make the filter a string.

Reply:
I'm not sure what you mean about how the severity filter should be an "equal" and not a "like" filter. It looks like it is an "equal" filter?

Reply:
The "bytes.Buffer" method was recommended as best practice upon research of concatonating strings (I also felt it was weird but when in Golang...)

For the filter, I expected it to work like "Where X = Y" not "Where X LIKE Y" as in partial match to the query is permitted (IE the => theme should not work with "=" or so my understanding went.

Reply:
i assume byte.Buffer is more efficient when concatenating a lot of strings, but i think for our purposes, its likely not actually much difference (as we're just concatenating ~10 strings at most)